packages/orchestration/src/examples/auto-stake-it-tap-kit.js

```diff
@@ -0,0 +1,148 @@
+import { M, mustMatch } from '@endo/patterns';
+import { E } from '@endo/far';
+import { VowShape } from '@agoric/vow';
+import { makeTracer } from '@agoric/internal';
+import { atob } from '@endo/base64';
+import { ChainAddressShape } from '../typeGuards.js';
+
+const trace = makeTracer('AutoStakeItTap');
+
+/**
+ * @import {VTransferIBCEvent} from '@agoric/vats';
+ * @import {VowTools} from '@agoric/vow';
+ * @import {Zone} from '@agoric/zone';
+ * @import {TargetApp} from '@agoric/vats/src/bridge-target.js';
+ * @import {ChainAddress, CosmosValidatorAddress, Denom, OrchestrationAccount} from '@agoric/orchestration';
+ * @import {FungibleTokenPacketData} from '@agoric/cosmic-proto/ibc/applications/transfer/v2/packet.js';
+ */
+
+/**
+ * @typedef {{
+ *   stakingAccount: ERef<OrchestrationAccount<{ chainId: 'cosmoshub' }>>;
```

Comment:
it won't always be `cosmoshub` but it will always have staking methods

```suggestion
 *   stakingAccount: ERef<OrchestrationAccount<any> & StakingAccountActions>;
```

Reply:
Thanks, didn't realize we could do this. As discussed ~yesterday, I hope there is room to simplify these generic parameters.